testgrid/cmd/configurator/prow.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"k8s.io/test-infra/testgrid/config"
+	"path"
+	"strings"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowConfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+	prowGCS "k8s.io/test-infra/prow/pod-utils/gcs"
+)
+
+const testgridTestGroupAnnotation = "testgrid-test-group"
```

Comment:
Is testgrid-test-group needed? Finding where the config is will already be a bit more confusing because it could be in the main config.yaml file, OR in any specified Prow config yaml, and it's a step less to find logs for a particular Prow job if the test group name has to match the job name.

(Plus there's not really a reason to specify the test group name custom in auto-generated config; it's under the Prow job and hopefully the Prow job's name will be descriptive enough to be useful).

Reply:
This is fair, I will remove it.

Reply:
I removed the ability to set the name, but left a `testgrid-create-test-group` boolean annotation, to force a test group to be created even if we aren't immediately adding it to a dashboard (because you configured that somewhere else). As before, specifying a dashboard implies you want a group created if necessary.

Thoughts?

Reply:
I think that's fine if necessary, but usually if there's a test group without a dashboard that means that it's eating up TestGrid resources, but no one is looking at them. I can't think of a use case for having a test group without a dashboard/dashboard tab, but I might be missing something?

Reply:
I think if you have a test group without a dashboard we fail it in presubmit (and that would still be true with these changes).

I was thinking more that the test groups would still be defined like this instead of expecting people   to remember how GCS works (I never can), but they might want fancier dashboard tab features and so have their tabs defined in config.yaml.

Reply:
Oh, gotcha! That seems fair then.